oem/ibm/libpldmresponder/file_io_type_lid.hpp

```diff
@@ -94,6 +107,49 @@ class LidHandler : public FileHandler
         return true;
     }
 
+    void validateMarkerLid(oem_platform::Handler* oemPlatformHandler)
+    {
+        if (oemPlatformHandler != nullptr)
+        {
+            pldm::responder::oem_ibm_platform::Handler* oemIbmPlatformHandler =
+                dynamic_cast<pldm::responder::oem_ibm_platform::Handler*>(
+                    oemPlatformHandler);
+            auto sensorId =
+                oemIbmPlatformHandler->codeUpdate->getMarkerLidSensor();
+            using namespace pldm::responder::oem_ibm_platform;
+            auto markerLidDirPath = fs::path(LID_STAGING_DIR) / "lid" /
+                                    markerLidName;
+            uint8_t validateStatus = VALID;
+            try
+            {
+                auto& bus = pldm::utils::DBusHandler::getBus();
+                auto method = bus.new_method_call(
+                    "xyz.openbmc_project.Software.BMC.Updater",
+                    "/xyz/openbmc_project/software",
+                    "xyz.openbmc_project.Software.LID", "Validate");
+                method.append(markerLidDirPath.c_str());
+                bus.call(method);
+            }
+            catch (const sdbusplus::exception::exception& e)
+            {
+                if (strcmp(e.name(), accessKeyExpired) == 0)
```

Comment:
strcmp() is a c style function , prefer c++ style

Reply:
done